app/src/main/java/com/team7/cmput301/android/theirisproject/IrisProjectApplication.java

```diff
@@ -92,6 +101,55 @@ public static Boolean isConnectedToInternet(Context context) {
 
     }
 
+    /**
+     * Make the update task if not already made.
+     *
+     * @param context The application context
+     */
+    public static void initBulkUpdater(Context context) {
+
+        if (updater == null) {
+
+            Callback<Boolean> cb = new Callback<Boolean>() {
+                @Override
+                public void onComplete(Boolean res) {
+                    handleBulkUpdateResult(res);
+                }
+            };
+
+            updater = new BulkUpdateTask(context, cb);
+
+        }
+
+    }
+
+    /**
+     * Put the specified model in appropriate update queue,
+     * and execute the updater task if not already running.
+     *
+     * @param model A model instance that wasn't successfully uploaded to elasticsearch
+     */
+    public static void putInUpdateQueue(Object model) {
+
+        if (model.getClass() == Problem.class) {
```

Comment:
Could probably use instanceof instead of comparing the classes but this is fine too